### PR TITLE
Opportunity/Event Card Section

### PR DESCRIPTION
**Pull request by jtcaovan**
# Pull Request Template

## Issue Overview

This PR addresses [#128](https://www.notion.so/exygy/Opportunity-Event-Cards-a5e5bba9d1d44c548a14ae85adb7a505?d=edd97c7c48414f849363058c2a0d0df4)

- [ ] This change addresses the issue in full
- [x] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR creates the Opportunity and Event card section for the home page. It pulls in content from sanity studio and styles them based on whether it is an event or opportunity card. Note: Section header, description, background color, and button text(+ styling) will be hardcoded in this PR and will be addressed in future tickets

## How Can This Be Tested/Reviewed?
[Figma Design](https://www.figma.com/design/iDBeNK7SnZXF8wc3pbPTCY/Our415-Site?node-id=2629-15679)

To view the sections, you can add the components in `Homepage.tsx`
 `<OppEventCardSection sectionType="opportunity"/>`
 `<OppEventCardSection sectionType="event"/>`

- [ ] Should pull 4 opportunities/events from Sanity studio, events shown in two by two layout and opps in four column layout
- [ ] Cards should be responsive, images should maintain aspect ratio on different screen sizes
- [ ] Opportunity/Event date should show `Month DD, YYYY - Month DD, YYYY` if start date is different from end date,  `Month DD, YYYY - 12:00PM - 1:00PM` if end time is included, or `Month DD, YYYY - 12:00PM `
- [ ] Cards should be clickable anywhere on the component while meeting accessibility requirements
- [ ] Review code

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I have assigned reviewers
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: app/components/ui/Section/OppEventCardSection.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      };
+      const query = `*[_type == "${sectionType}"] | order(_createdAt desc)[0...4] { ${fields[sectionType]} }`;
+
+      const result: EventData[] = await client.fetch(query);
+      setEventData(result);
+    };
+
+    setIsLoading(true);
+    fetchEventData();
+    setIsLoading(false);
```
Comment (kramduckner) [inline app/components/ui/Section/OppEventCardSection.tsx:66]: I think setIsLoading will be set to false before the event data finishes fetching. I believe if you call setIsLoading(false) inside the fetching function after the fetch, it'll render the states in the proper order!

File: app/components/ui/Cards/OppEventCard.module.scss [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    padding: 24px;
+    height: 100%;
+    width: 100%;
+
+    @media screen and (max-width: $break-tablet-p) {
+      padding: 16px;
+    }
+
+    .contentTitle {
+      font-size: 20px;
```
Comment (kramduckner) [inline app/components/ui/Cards/OppEventCard.module.scss:49]: A lot of the font styling gets set in _helpers.scss. Mind removing the styling you added that gets set there? Just want to try to keep the 'theme' functionality!

**Review by kramduckner — CHANGES_REQUESTED** [review]
Looking good! 

Do you mind refactoring a bit of the styling to use those scss tokens (there's a comment going into a bit more detail) and removing some of the styling that gets set in _helpers.scss? Other than that and addressing the loading state, it's ready for merge imo!

File: app/components/ui/Cards/OppEventCard.module.scss [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    height: 100%;
+    width: 100%;
+
+    @media screen and (max-width: $break-tablet-p) {
+      padding: $spacing-4;
+    }
+
+    .contentTitle {
+      line-height: 125%;
+      margin-bottom: $spacing-4;
```
Comment (kramduckner) [inline app/components/ui/Cards/OppEventCard.module.scss:50]: can we use the $general-spacing-md here? I'd love to use those general spacing vars as much as possible!

**Review by kramduckner — CHANGES_REQUESTED** [review]
Looking great! Just one last nit - can you try to use those $general-spacing vars as much as possible?

File: app/components/ui/Cards/OppEventCard.module.scss (hunk 3 above)
Comment (jtcaovan) [reply app/components/ui/Cards/OppEventCard.module.scss]: thanks for pointing that out!

Comment (kramduckner) [reply app/components/ui/Cards/OppEventCard.module.scss]: thank you!

**Review by kramduckner — APPROVED** [review]

**Pull request merged**